expyfun/_utils.py

```diff
@@ -426,12 +418,21 @@ def verbose_dec(function, *args, **kwargs):
         return ret
 
 
+def _has_avbin():
+    try:
+        from pyglet.media.avbin import AVbinSource
+        del AVbinSource
+        return True
+    except ImportError:
+        return False
+
+
 _is_appveyor = (os.getenv('APPVEYOR', 'False').lower() == 'true')
 requires_pandas = skipif(has_pandas is False, 'Requires pandas')
 requires_h5py = skipif(has_h5py is False, 'Requires h5py')
 requires_joblib = skipif(has_joblib is False, 'Requires joblib')
 requires_opengl21 = skipif(_is_appveyor, 'Appveyor OpenGL too old')
-requires_avbin = skipif(has_avbin is False, 'Requires AVbin')
+requires_avbin = skipif(_has_avbin() is False, 'Requires AVbin')
```

Comment:
This will still be run on import. You will need to do something more like this, which only gets called when the decorator is accessed:

https://github.com/mne-tools/mne-python/blob/master/mne/utils.py#L680